### PR TITLE
Add NavLink option to use another element type for active state

### PR DESCRIPTION
**Pull request by ebednarz**
I was a bit disappointed that the NavLink component active state produces a href attribute with a same page reference. Besides potential unexpected activation side effects, that's fairly bad UX, especially when using the keyboard (it's possible to fix the wrong problem with "pointer-events: none" for pointing devices, but I don't see a way to do that to the tab index).

Would you be interested in adding an optional activeElement property? It might be preferable to [remove the "href" attribute](http://www.w3.org/TR/html5/text-level-semantics.html#the-a-element), but unfortunately [that's not currently possible with React](https://github.com/facebook/react/issues/1448).

I tried my best to make the code change simple, not smart. :-)

### CODE REVIEW

File: tests/unit/lib/NavLink-test.js [hunk 1]
```diff
@@ -101,6 +101,15 @@ describe('NavLink', function () {
...
             expect(link.getDOMNode().getAttribute('class')).to.equal('active');
         });
+        it('should set active state by tag name if the optional activeElement property is set', function () {
+            var navParams = {a: 1, b: 2};
+            var link = ReactTestUtils.renderIntoDocument(
+                <MockAppComponent context={mockContext}>
+                    <NavLink activeElement="span" routeName='foo' />
+                </MockAppComponent>
+            );
+            expect(link.getDOMNode().nodeName.toLowerCase()).to.equal('span');
```
Comment (roderickhsiao) [inline tests/unit/lib/NavLink-test.js]: https://facebook.github.io/react/docs/top-level-api.html#react.finddomnode

**Comment by mridgway** [comment]
@ebednarz I haven't heard of this use case before. Can you explain why it's bad UI to have a link for the current page? A lot of times I see this as a way of refreshing the current page.

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by ebednarz** [comment]
@mridgway It often leads to errors and superfluous network requests. Inability to unlink the current resource is simply considered a technical shortcoming where I come from, not a design choice.
I can't link to work briefings, but this sums it up nicely:
 http://www.foresee.com/usability-best-practices-self-referencing-links/

**Comment by ebednarz** [comment]
@mridgeway Any new insights on this topic? If there's anything I can do, I'd be happy to.

File: tests/unit/lib/NavLink-test.js (hunk 1 above)
Comment (ebednarz) [inline tests/unit/lib/NavLink-test.js]: That would apply to all tests, not my pull request in particular. I like to be consistent. :)

**Comment by ebednarz** [comment]
@mridgway FYI, I merged master into the pull request. (I also created a fork for npm installation for the time being.)

**Pull request merged**